### ISSUE TITLE
Using with keyed each blocks?

### ISSUE THREAD

**Issue by jacobpake** [orefalo/svelte-splitpanes#35]
Looking for some advice for using this library with keyed each blocks:

```svelte
<Splitpanes>
  {#each array as object (object.id)}
    <Pane>
      <InnerComponent value={object.value}/>
    </Pane>
  {/each}
</Splitpanes>

```

I want to change the order of the array (and resulting panes) without re-rendering the InnerComponent. 

When I use the above code, the InnerComponent does not re-render (good). 

Neither do the Panes (bad): the splitters act as if the Pane was in it's original position (e.g. resizing effects the wrong Pane).

**Comment by orefalo** [comment]
Hi Jacob, thank you - may I ask you to compile a REPL https://svelte.dev/repl/hello-world?version=3.49.0

if i get you right it's a refresh issue, isn't it?

**Comment by Tal500** [comment]
Hi @jacobpake !

While this is a valid use case, I think no one thought about it.

We have in the library two problems:
(1) The library was designed to handle panes add and remove, but not reordering.
(2) Even in the case of panes add/removal, for simplicity the sizes of the panes get reset, and not saved.

I think we can fix (1) by checking the location of the pane element in the DOM in runtime.
But even if you check its location during the mouse dragging process, it still problematic in other parts, like the optional first splitter and a11y(planned in the future).
We need to think how we can notice that the order get changed. Maybe create a dedicated component that simulates keyed each block with some slots magic, and notify the library when the order gets changed? Maybe the Svelte [`afterUpdate` event](https://svelte.dev/tutorial/update) gets triggered when the order is changed? (which gives a simple solution)

Problem (2) should be fixed in a wider context than here. As a temporary hack, you can bind the panes size to a variable (except the last one who should be `100 - (the rest)`), and then the sizes will be saved.

**Comment by Tal500** [comment]
> Problem (2) should be fixed in a wider context than here. As a temporary hack, you can bind the panes size to a variable (except the last one who should be `100 - (the rest)`), and then the sizes will be saved.

Update: Problem 2 was fixed by a7ab5b3e43729cdeca76410f4e8f2e524e817632 and 24ac461d855fc9c986e72fef62cd0b16311d653e.

**Comment by Tal500** [comment]
> We have in the library two problems: (1) The library was designed to handle panes add and remove, but not reordering. (2) Even in the case of panes add/removal, for simplicity the sizes of the panes get reset, and not saved.
> 
> I think we can fix (1) by checking the location of the pane element in the DOM in runtime. But even if you check its location during the mouse dragging process, it still problematic in other parts, like the optional first splitter and a11y(planned in the future). We need to think how we can notice that the order get changed. Maybe create a dedicated component that simulates keyed each block with some slots magic, and notify the library when the order gets changed? Maybe the Svelte [`afterUpdate` event](https://svelte.dev/tutorial/update) gets triggered when the order is changed? (which gives a simple solution)

My internal tests shows that Svelte `beforeUpdate` is fired in `Splitpanes` component when this reordering happen, so we can listen to this event and see if the panes in the DOM are in the correct order. If not, it means we need to update the internal order saved by the library.

It is much easier now to fix
